lib/logstash/outputs/s3.rb

```diff
@@ -335,20 +383,29 @@ def shutdown_upload_workers
   end
 
   private
-  def handle_event(encoded_event)
+  def handle_event(encoded_event, event)
+    actualprefix = event.sprintf(@prefix)
+    if not @prefixes.to_a().include? actualprefix
+       @file_rotation_lock[actualprefix] = Mutex.new
+       @prefixes.add(actualprefix)
+       reset_page_counter(actualprefix)
+       create_temporary_file(actualprefix)       
+       @empty_uploads[actualprefix] = 0       
+    end
```

Comment:
Might be worth it to move all this logic into his own method? What do you think?

`initialize_prefix(prefix)` or `configure_prefix(prefix)`